### PR TITLE
octoprint: 1.4.2 -> 1.5.1

### PR DESCRIPTION
**Pull request by lovesegfault**
###### Motivation for this change
I noticed it was out of date, the update required me to shave some yaks as well

- python3Packages.sgmllib3k: init at 1.0.0
- python3Packages.feedparser: 5.2.1 -> 6.0.2
- octoprint: 1.4.2 -> 1.5.1
- octoprint: sort propagatedBuildInputs

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/sgmllib3k/default.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  pname = "sgmllib3k";
+  version = "1.0.0";
+  disabled = isPy27;
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "1s8jm3dgqabgf8x96931scji679qkhvczlv3qld4qxpsicfgns3q";
+  };
+
+  # N.B. Tests are completely broken. The package doesn't even ship with the
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/sgmllib3k/default.nix:16]: Did you try the link from the upstream repo?

Comment (lovesegfault) [reply pkgs/development/python-modules/sgmllib3k/default.nix:16]: What do you mean? I couldn't even find the upstream repo, only the PyPI tarball

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 105913` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>7 packages failed to build:</summary>
  <ul>
    <li>appdaemon</li>
    <li>quodlibet</li>
    <li>quodlibet-full</li>
    <li>quodlibet-without-gst-plugins</li>
    <li>quodlibet-xine</li>
    <li>quodlibet-xine-full</li>
    <li>rss2email</li>
  </ul>
</details>
<details>
  <summary>23 packages built:</summary>
  <ul>
    <li>calibre (calibre-py3)</li>
    <li>canto-curses</li>
    <li>canto-daemon</li>
    <li>flexget</li>
    <li>foxtrotgps</li>
    <li>gpodder</li>
    <li>greg</li>
    <li>kmymoney</li>
    <li>octoprint</li>
    <li>pubs</li>
    <li>python37Packages.feedparser</li>
    <li>python37Packages.sgmllib3k</li>
    <li>python37Packages.weboob</li>
    <li>python38Packages.feedparser</li>
    <li>python38Packages.sgmllib3k</li>
    <li>python38Packages.weboob</li>
    <li>python39Packages.feedparser</li>
    <li>python39Packages.sgmllib3k</li>
    <li>python39Packages.weboob</li>
    <li>sabnzbd</li>
    <li>tartube</li>
    <li>tribler</li>
    <li>ytcc</li>
  </ul>
</details>

File: pkgs/development/python-modules/sgmllib3k/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/development/python-modules/sgmllib3k/default.nix:16]: I meant to ask if we could fetch the source code from GitHub or somewhere else. The repo appears to be https://github.com/hsoft/sgmllib.

Comment (orivej) [reply pkgs/development/python-modules/sgmllib3k/default.nix]: I have switched the source to github, now the tests pass.

Comment (lovesegfault) [reply pkgs/development/python-modules/sgmllib3k/default.nix]: Nice, thank you!

**Comment by lovesegfault** [comment]
This currently fixes the broken feedparser in nixpkgs master. cc. @FRidh @jonringer

**Comment by orivej-nixos** [comment]
Thanks! I have fixed various issue that prevented building octoprint on the current master, and now it builds again.

**Pull request merged**